File: Program's_Contributed_By_Contributors/Java/Sorting Algos/SelectionSort.java

import java.util.*;

public class SelectionSort {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		int[] arr = { 4, 6, 7, 2, 1, 5, 3, 8, 9 };
		selectionSorting(arr);
		System.out.println(Arrays.toString(arr));

	}

	static void selectionSorting(int[] arr) {
		for (int i = 0; i < arr.length; i++) {
//			find the max item in the remaining array and swap with correct index

			int last = arr.length - i - 1;

			int maxInd = getMaxInd(arr, 0, last);
			swapArray(arr, maxInd, last);
		}
	}

	private static int getMaxInd(int[] arr, int start, int end) {
		int max = start;

		for (int i = start; i <= end; i++) {
			if (arr[max] < arr[i])
				max = i;
		}
		return max;
	}

	static void swapArray(int[] arr, int first, int second) {
		int temp = arr[first];
		arr[first] = arr[second];
		arr[second] = temp;
	}

}

File: Program's_Contributed_By_Contributors/C++ Programs/sorting/selectionSort.cpp
#include <iostream>
using namespace std;

void swap(int *x, int *y) {
    int temp = *x;
    *x = *y;
    *y = temp;
}

void selectionSort(int arr[], int n) {
    int i, j, max;
    for (i = n-1; i>0; i--) {
        max = 0;
        for (j = 1; j <= i; j++)
            if (arr[j] > arr[max])
                max = j;
        if (max != i)
            swap(&arr[max], &arr[i]);

    }
}

void show(int arr[], int n) {
    for (int i = 0; i < n; i++)
        cout << arr[i] << ' ';
    cout << endl;
}

int main() {
    int arr[] = {64, 34, 25, 12, 22, 11, 90};
    int n = sizeof(arr) / sizeof(arr[0]);
    cout << "Before \t";
    show(arr, n);
    selectionSort(arr, n);
    cout << "Sorted array: \t";
    show(arr, n);
    return 0;
}